### ISSUE TITLE
`$R` variable isn't set in JAFFA_stage.groovy file ?

### ISSUE THREAD

**Issue by serine** [Oshlack/JAFFA#45]
Hi there, 

I'm having various issues with successfully finishing `JAFFA` run. Currently it is falling over at the step where is uses Rscripts. 

I ran this script `process_transcriptome_blat_table.R` manually, but then it fell over in the later step. 

I just want to clarify that `JAFFA_stage.groovy` doesn't have a bug. [This line](https://github.com/Oshlack/JAFFA/blob/c2ebffdeb2b0ad3518fb77a3e966a10c28760507/JAFFA_stages.groovy#L364) uses `$R` variable. I'm assuming that needs to be 

```
R = RScript
```
but that variable isn't set any where in the file

```
grep ^R JAFFA_stages.groovy 
R_filter_transcripts_script=codeBase+"/process_transcriptome_blat_table.R"
R_get_final_list=codeBase+"/make_final_table.R"
R_get_spanning_reads_script=codeBase+"/get_spanning_reads.R"
R_get_spanning_reads_direct_script1=codeBase+"/get_spanning_reads_for_direct_1.R"
R_get_spanning_reads_direct_script2=codeBase+"/get_spanning_reads_for_direct_2.R"
R_compile_results_script=codeBase+"/compile_results.R"
```

but evidently used through the file 

```
grep "\$R" JAFFA_stages.orig 
                time $R --vanilla --args $input $output $gapSize $transTable <
                    $R_filter_transcripts_script &> ${output.dir}/log_filter
               $R --vanilla --args ${output.dir}/${branch} $input1 ${output.dir}/${branch}.temp 
                   $output ${output.dir}/${branch}.readLengths $overHang < $R_get_spanning_reads_script ;
                $R --no-save --args $input1 $transTable ${output.dir}/temp_gene_ids ${output.dir}/paired_contigs.temp
                    < $R_get_spanning_reads_direct_script1 ;
                $R --no-save --args $input1 ${output.dir}/spanning_pair_counts.temp $output < $R_get_spanning_reads_direct_script2 ;
                $R --vanilla --args $input1 $input2 $transTable $knownTable $finalGapSize $exclude $output < $R_get_final_list
            $R --vanilla --args $outputName $inputs < $R_compile_results_script ;
```

Basically I'm asking is it a bug i.e I need to add `R = RScript` in OR that `R` variable gets set somewhere later in the run. If it is the later, how does it get set

Thanks,

**Comment by nadiadavidson** [comment]
Hi,
This variable gets set in the file, tools.groovy, and should be set to the R command eg. could be something like 
R=R or R=<path to R>
If you send some more info about the error you're getting I might be able to help.

Cheers,
Nadia.

**Comment by serine** [comment]
right, makes a bit more sense with that input redirect of the script. 

I'm guessing `tools.groovy` automatically generated on the install? 

```// Path to tools used by the JAFFA pipeline
bpipe="/home/kirill/downloads/JAFFA/tools/bin/bpipe"
velveth="/home/kirill/downloads/JAFFA/tools/bin/velveth"
velvetg="/home/kirill/downloads/JAFFA/tools/bin/velvetg"
oases="/home/kirill/downloads/JAFFA/tools/bin/oases"
trimmomatic="/home/kirill/downloads/JAFFA/tools/bin/trimmomatic"
samtools="/home/kirill/downloads/JAFFA/tools/bin/samtools"
bowtie2="/home/kirill/downloads/JAFFA/tools/bin/bowtie2"
blat="/home/kirill/downloads/JAFFA/tools/bin/blat"
dedupe="/home/kirill/downloads/JAFFA/tools/bin/dedupe"
reformat="/home/kirill/downloads/JAFFA/tools/bin/reformat"
R=""
```

`R` wasn't load in, hence why I'm guessing it didn't get set

slightly aside, just for my own curiously, is much advantage with input redirect rather then 

```   from(".psl") {
            exec """
                time $R --vanilla $R_filter_transcripts_script $input \
                                                               $output \
                                                               $gapSize \
                                                               $transTable &> ${output.dir}/log_filter
            ""","filter_transcripts"
        }
```

assuming `R = Rscript`. or just the matter of taste. I was rather confused by that input redirect at the start. 

thanks for responding so quickly

**Comment by serine** [comment]
ok, I think I got it now, it is in the docs as well

```
       R [options] [< infile] [> outfile]
```

thanks
